### PR TITLE
Location addresses & Bug fixes

### LINKED ISSUE #224: Addresses for Locations
We need to implement the adding and viewing of physical addresses for "Locations" in the UI - I believe this was accounted for in the DB, but the UI doesn't have it yet. I would like to understand what it would take to implement this - since there will be students working on this next semester, perhaps it is something they can do and therefore not something that has to get done in the next few weeks.

### PR DESCRIPTION
**Pull request by blackjack26**
- Added addresses for location
- Editing addresses
- Deleting addresses
- Fixed intake form view / edit
- Fixed self-referral view / edit
- Fixed referral view / edit

Fixes #224

### CODE REVIEW

File: views/locations/locations_modify.php [hunk 1]
```diff
@@ -37,22 +39,78 @@
...
+
+    // Logic for parsing the address into the address number and street name.
+    $intake_street_num = NULL;
+    $intake_street_name = NULL;
+
+    if ($street_address !== '') {
+        $intake_address_info = explode(" ", $street_address);
+
+        // Loop to parse through the address array ($self_address_info)
+        for($i = 0; $i < sizeOf($intake_address_info); $i++){
```
Comment (michelleburd) [inline views/locations/locations_modify.php]: Formatting 🤢

File: views/locations/locations_delete.php [hunk 2]
```diff
@@ -46,11 +46,17 @@
         if ($success) {
             $deleteRes = $db->query("DELETE FROM facilitatorclassattendance WHERE sitename = $1", [$sitename]);
             $success = $deleteRes && pg_result_error_field($deleteRes, PGSQL_DIAG_SQLSTATE) == 0;
+        }
```
Comment (ghost) [inline views/locations/locations_delete.php]: Should also delete from ClassOffering, which would be necessary if  `hasRole(Role::Superuser)` and `$notConnected` is false

**Review by ghost — APPROVED** [review]
looks good to me.

**Review by christianmenk — APPROVED** [review]
looks good to me as well. the queries look much cleaner than before.

**Review by michelleburd — APPROVED** [review]

**Pull request merged**